src/Convert.hs

```diff
@@ -21,17 +22,16 @@ createArguments mp3Directory filePath =
 getMP4FilesFromPath :: Turtle.Text -> Shell Turtle.FilePath
 getMP4FilesFromPath filepath = find (suffix ".mp4") $ Turtle.fromText filepath
 
-getDirectories :: Reader Config (Turtle.Text, Turtle.Text)
-getDirectories = do
-  mp4Directory <- asks (lookUpConfig "MP4_DIRECTORY")
-  mp3Directory <- asks (lookUpConfig "MP3_DIRECTORY")
-  return (mp4Directory, mp3Directory)
-
 runCommand :: Turtle.Text -> Turtle.FilePath -> IO ()
 runCommand mp3Directory filePath = proc "ffmpeg" (createArguments mp3Directory filePath) Turtle.empty >>= print
 
+appConvert :: ReaderT Config IO ()
+appConvert = do
+  eitherMP4Directory <- asks (lookUpConfig "MP4_DIRECTORY")
+  eitherMP3Directory <- asks (lookUpConfig "MP3_DIRECTORY")
+  case (eitherMP4Directory, eitherMP3Directory) of
+    (Right mp4Dir, Right mp3Dir) -> foldIO (getMP4FilesFromPath mp4Dir) (L.sink (runCommand mp3Dir))
+    (_, _) -> return () -- putStrLn "Error " was not working
```

Comment:
Still haven't quite figured out why I couldn't do a `putStrLn "Error"` even though that returned me a `IO ()`